risc0/bigint2/src/ec/curve.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::EC_256_WIDTH_WORDS;
+
+/// The secp256k1 curve's prime as u32 digits, least significant digit first
+const SECP256K1_PRIME: [u32; EC_256_WIDTH_WORDS] = [
+    0xFFFFFC2F, 0xFFFFFFFE, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF,
+];
+const SECP256K1_CURVE: &WeierstrassCurve<EC_256_WIDTH_WORDS> =
+    &WeierstrassCurve::<EC_256_WIDTH_WORDS>::new(
+        SECP256K1_PRIME,
+        [0u32; EC_256_WIDTH_WORDS],
+        [7, 0, 0, 0, 0, 0, 0, 0],
+    );
+
+/// Generic static curve configuration.
+pub trait Curve<const WIDTH: usize> {
+    const CURVE: &'static WeierstrassCurve<WIDTH>;
+}
+
```

Comment:
I guess I was thinking something like a generic Curve trait could live in `ec` but then we'd have a separate module for each kind of specific curve.

Reply:
Sure! Was this what you had in mind? [`abccf42` (#2572)](https://github.com/risc0/risc0/pull/2572/commits/abccf42fe4e617774e91706ad5bdf1d406ff7fe8)

Or did you still want the curve types kept in their same module for organization? Exports are the same either way